hs_rest_api2/serializers.py

```diff
@@ -0,0 +1,172 @@
+
+from hsmodels.schemas.resource import ResourceMetadataIn
+from hsmodels.schemas.aggregations import GeographicFeatureMetadataIn, GeographicRasterMetadataIn, \
+    MultidimensionalMetadataIn, SingleFileMetadataIn, FileSetMetadataIn, TimeSeriesMetadataIn, \
+    ReferencedTimeSeriesMetadataIn, ModelProgramMetadata, ModelProgramMetadataIn, ModelInstanceMetadata, \
+    ModelInstanceMetadataIn
+
+
+class ResourceMetadataInForbidExtra(ResourceMetadataIn):
+
+    class Config:
+        extra = 'forbid'
+
+
+class ResourceMetadataSerializer(Serializer):
+    class Meta:
+        fields = "__all__"
+        swagger_schema_fields = ResourceMetadata.schema()
+
+
+class ResourceMetadataInSerializer(Serializer):
+    class Meta:
+        fields = "__all__"
+        _schema = ResourceMetadataInForbidExtra.schema()
+        _schema['title'] = _schema['title'] + " In"
+        swagger_schema_fields = _schema
+
+
+class GeographicFeatureMetadataSerializer(Serializer):
+    class Meta:
+        fields = "__all__"
+        swagger_schema_fields = GeographicFeatureMetadata.schema()
+
+
+class GeographicFeatureMetadataInSerializer(Serializer):
+    class Meta:
+        fields = "__all__"
+        _schema = GeographicFeatureMetadataIn.schema()
+        _schema['title'] = _schema['title'] + " In"
+        swagger_schema_fields = _schema
+
+
+class GeographicRasterMetadataSerializer(Serializer):
+    class Meta:
+        fields = "__all__"
+        swagger_schema_fields = GeographicRasterMetadata.schema()
+
+
+class GeographicRasterMetadataInSerializer(Serializer):
+    class Meta:
+        fields = "__all__"
+        _schema = GeographicRasterMetadataIn.schema()
+        _schema['title'] = _schema['title'] + " In"
+        swagger_schema_fields = _schema
+
+
+class MultidimensionalMetadataSerializer(Serializer):
+    class Meta:
+        fields = "__all__"
+        swagger_schema_fields = MultidimensionalMetadata.schema()
+
+
+class MultidimensionalMetadataInSerializer(Serializer):
+    class Meta:
+        fields = "__all__"
+        _schema = MultidimensionalMetadataIn.schema()
+        _schema['title'] = _schema['title'] + " In"
+        swagger_schema_fields = _schema
+
+
+class SingleFileMetadataSerializer(Serializer):
+    class Meta:
+        fields = "__all__"
+        swagger_schema_fields = SingleFileMetadata.schema()
+
+
+class SingleFileMetadataInSerializer(Serializer):
+    class Meta:
+        fields = "__all__"
+        _schema = SingleFileMetadataIn.schema()
+        _schema['title'] = _schema['title'] + " In"
+        swagger_schema_fields = _schema
+
+
+class FileSetMetadataSerializer(Serializer):
+    class Meta:
+        fields = "__all__"
+        swagger_schema_fields = FileSetMetadata.schema()
+
+
+class FileSetMetadataInSerializer(Serializer):
+    class Meta:
+        fields = "__all__"
+        _schema = FileSetMetadataIn.schema()
+        _schema['title'] = _schema['title'] + " In"
```

Comment:
It's just a label difference for the OpenAPI (swagger) interface.  Otherwise, the schemas would have the same titles for the in/out schemas.
<img width="465" alt="image" src="https://user-images.githubusercontent.com/35820390/145256948-0e67ade5-1151-468e-9a5a-0a92b0717efa.png">